app/controllers/course/assessment/question/scribing_controller.rb

```diff
@@ -0,0 +1,155 @@
+                              through: :assessment, parent: false
+
+  include CarrierWave::MiniMagick
+
+  def new
+    respond_to do |format|
+      format.html { render 'new' }
+    end
+  end
+
+  def show
+    respond_to do |format|
+      format.json { render_scribing_question_json }
+    end
+  end
+
+  def create # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    # special handling for PDF files
+    if params&.[](:question_scribing)&.[](:file)&.content_type&.downcase == 'application/pdf'
+      file = params[:question_scribing][:file]
+      respond_to do |format|
+        if generate_pdf_questions(file)
+          format.json { render_success_json t('.success') }
+        else
+          format.json { render_failure_json t('.failure') }
+        end
+      end
+    else
+      respond_to do |format|
+        if @scribing_question.save
+          format.json { render_scribing_question_json }
+        else
+          format.json { render_failure_json t('.failure') }
+        end
+      end
+    end
+  end
+
+  def edit
+    respond_to do |format|
+      format.html { render 'edit' }
+      format.json { render_scribing_question_json }
+    end
+  end
+
+  def update
+    # Don't allow the user to edit attachment at the moment
+    # In case the user uploads a multipage pdf to replace
+    respond_to do |format|
+      if @scribing_question.update(scribing_question_update_params)
+        format.json { render_scribing_question_json }
+      else
+        format.json { render_failure_json t('.failure') }
+      end
+    end
+  end
+
+  def destroy
+    if @scribing_question.destroy
+      redirect_to course_assessment_path(current_course, @assessment),
+                  success: t('.success')
+    else
+      error = @scribing_question.errors.full_messages.to_sentence
+      redirect_to course_assessment_path(current_course, @assessment),
+                  danger: t('.failure', error: error)
+    end
+  end
+
+  private
+
+  def render_scribing_question_json
+    render partial: 'scribing_question', locals: { scribing_question: @scribing_question }
+  end
+
+  def scribing_question_params
+    params.require(:question_scribing).permit(
+      :title, :description, :staff_only_comments, :maximum_grade,
+      attachment_params, skill_ids: []
+    )
+  end
+
+  # Don't allow uploading of files at the moment
+  def scribing_question_update_params
+    params.require(:question_scribing).permit(
+      :title, :description, :staff_only_comments, :maximum_grade,
+      skill_ids: []
+    )
+  end
+
+  def render_success_json(message)
+    render json: { message: message },
+           status: :ok
+  end
+
+  def render_failure_json(message)
+    render json: { message: message, errors: @scribing_question.errors.full_messages },
+           status: :bad_request
+  end
+
+  def generate_pdf_questions(file) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
```

Comment:
I think you can move this into a service. Easier to manage and refactor when the logic is in a separate file.